File: src/types/timeline.ts
export interface VideoClip {
  id: string;
  source: string;
  start_ms: number;
  end_ms: number;
  trim_start_ms: number;
  trim_end_ms: number;
}

export interface AudioClip {
  id: string;
  source: string;
  start_ms: number;
  end_ms: number;
  volume: number;
}

File: src-tauri/src/models/timeline.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TimelineTrack {
    pub id: String,
    #[serde(rename = "type")]
    pub track_type: TrackType,
    pub clips: Vec<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TrackType {
    Video,
    Telop,
    Audio,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VideoClip {
    pub id: String,
    pub source: String,
    pub start_ms: u64,
    pub end_ms: u64,
    pub trim_start_ms: u64,
    pub trim_end_ms: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AudioClip {
    pub id: String,
    pub source: String,
    pub start_ms: u64,
    pub end_ms: u64,
    pub volume: f32,
}
